Modules/Loadable/Segmentations/Widgets/qMRMLSegmentsTableView.cxx

```diff
@@ -460,6 +469,76 @@ void qMRMLSegmentsTableView::onSegmentsTableClicked(const QModelIndex& modelInde
   }
 }
 
+//-----------------------------------------------------------------------------
+void qMRMLSegmentsTableView::onSegmentsTableDoubleClicked(const QModelIndex& modelIndex)
+{
+  Q_D(qMRMLSegmentsTableView);
+  QString segmentId = d->SortFilterModel->segmentIDFromIndex(modelIndex);
+  QStandardItem* item = d->Model->itemFromSegmentID(segmentId);
+  if (!d->SegmentationNode || !item)
+  {
+    return;
+  }
+
+  Qt::ItemFlags flags = item->flags();
+  if (!flags.testFlag(Qt::ItemIsSelectable))
+  {
+    return;
+  }
+
+  vtkSegment* segment = d->SegmentationNode->GetSegmentation()->GetSegment(segmentId.toStdString());
+  if (modelIndex.column() == d->Model->colorColumn())
+  {
+    if (QSettings().value(d->TerminologySelectorSettingsKey).toBool() == true)
+    {
+      vtkSlicerTerminologiesModuleLogic* terminologiesLogic = vtkSlicerTerminologiesModuleLogic::SafeDownCast(
+        qSlicerCoreApplication::application()->moduleLogic("Terminologies"));
+      if (!terminologiesLogic)
+      {
+        return;
+      }
+
+      // Get terminology from segment. Do not check success, as an empty terminology is also a valid starting point
+      vtkNew<vtkSlicerTerminologyEntry> terminologyEntry;
+      std::string serializedTerminology("");
+      if (segment->GetTag(vtkSegment::GetTerminologyEntryTagName(), serializedTerminology))
+      {
+        terminologiesLogic->DeserializeTerminologyEntry(serializedTerminology.c_str(), terminologyEntry);
```

Comment:
Please check if everything works without calling

    // Get default color and other metadata from loaded terminologies
    logic->UpdateEntryFromLoadedTerminologies(terminologyEntry);

You need to call `UpdateEntryFromLoadedTerminologies` if you need any information that was not in the `serializedTerminology` string.

Reply:
It is only for identification, so only the serialized components are needed. I did try it and it works.